pyproject.toml

```diff
@@ -7,43 +7,133 @@ license = "MIT"
+memray = "^1.3.1"
+pyinstrument = "^4.3.0"
+rich = "^11.2.0"
+sphinx = "^5.3.0"
+sphinxcontrib-programoutput = "^0.17"
 
 [tool.poetry.scripts]
-sphinx-performance = 'sphinx_performance.performance:cli_performance'
 sphinx-analysis = 'sphinx_performance.analysis:cli_analysis'
+sphinx-performance = 'sphinx_performance.performance:cli_performance'
+
+[tool.ruff]
+select = ["ALL"] # Enable all checks and maintain an ignore list
+# a lot of ignores here due to activation of ruff linting in an established project
+# candidates for reactivation are marked with a star
+ignore = [
+    "ANN001",  # missing-type-function-argument *
+    "ANN003",  # missing-type-kwargs *
+    "ANN101",  # missing-type-self
+    "ANN201",  # missing-return-type-undocumented-public-function *
+    "ANN202",  # missing-return-type-private-function *
+    "B905",    # zip-without-explicit-strict *
+    "C901",    # complex-structure *
+    "D100",    # undocumented-public-module *
+    "D101",    # undocumented-public-class *
+    "D102",    # undocumented-public-method *
+    "D107",    # no docstring needed for __init__
+    "D203",    # D203 and D211 are mutually exclusive: https://github.com/PyCQA/pydocstyle/issues/141
+    "D212",    # D212 and D213 are mutually exclusive: https://stackoverflow.com/a/45990465
+    "FBT002",  # boolean-default-value-positional-argument *
+    "PERF401", # manual-list-comprehension *
+    "PLR0912", # too-many-branches *
+    "PLR0913", # too-many-arguments *
+    "PLR0915", # too-many-statements *
+    "PLW0603", # global-statement *
+    "PLW1510", # subprocess-run-without-check *
+    "S603",    # subprocess-without-shell-equals-true *
+    "S607",    # start-process-with-partial-path *
+]
+fixable = ["ALL"]
+
+# Allow autofix for all enabled rules (when `--fix`) is provided.
+# fixable = ["A", "B", "C", "D", "E", "F", "..."]
+# unfixable = []
+
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".hg",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "venv",
+    ".venv",
+]
+
+# a bit longer than black for e.g. long docstrings that black did not shorten
+line-length = 100
+
+# Allow unused variables when underscore-prefixed.
+dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
 
-[tool.poetry.extras]
-docs = ["sphinx"]
+# Assume Python 3.11.
+target-version = "py311"
+
+[tool.ruff.pydocstyle]
+convention = "pep257"
+
+[tool.ruff.mccabe]
+max-complexity = 20
+
+[tool.ruff.per-file-ignores]
+"sphinx_performance/projects/basic/performance.py" = ["INP001"] # template dir
+"sphinx_performance/projects/needs/performance.py" = ["INP001"] # template dir
+"sphinx_performance/projects/theme/performance.py" = ["INP001"] # template dir
+
+[tool.black]
+target-version = ["py311"]
```

Comment:
As above, Should this not be the lowest supported version, i.e. new syntaxes in python 3.11 may not be supported by 3.8